plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/TransactionDirectoryListingCacheKey.java

```diff
@@ -28,21 +29,29 @@ public class TransactionDirectoryListingCacheKey
 
     private final long transactionId;
     private final Location path;
+    private final SchemaTableName schemaTableName;
 
-    public TransactionDirectoryListingCacheKey(long transactionId, Location path)
+    public TransactionDirectoryListingCacheKey(long transactionId, Location path, SchemaTableName schemaTableName)
     {
         this.transactionId = transactionId;
         this.path = requireNonNull(path, "path is null");
+        this.schemaTableName = requireNonNull(schemaTableName, "schemaTableName is null");
     }
 
     public Location getPath()
     {
         return path;
     }
 
+    public SchemaTableName getSchemaTableName()
+    {
+        return schemaTableName;
+    }
+
     public long getRetainedSizeInBytes()
     {
-        return INSTANCE_SIZE + estimatedSizeOf(path.toString());
+        return INSTANCE_SIZE +
+               estimatedSizeOf(path.toString()) + schemaTableName.getRetainedSizeInBytes();
```

Comment:
Nit: let's line break for each element being added

Reply:
of course!